File: frontend/src/types/game.ts
import { z } from 'zod'
import { GameListItemSchema } from '@/types/userGame'

export const GameReadSchema = GameListItemSchema.extend({
  summary: z.string().nullable(),
  storyline: z.string().nullable(),
  developer: z.string().nullable(),
  steam_description: z.string().nullable(),
  steam_reviews_summary: z.string().nullable(),
  hltb_extra: z.number().nullable(),
  hltb_completionist: z.number().nullable(),
  opencritic_excerpts: z.array(z.string()).nullable(),
  keywords: z.array(z.string()).nullable(),
  modes: z.array(z.object({ id: z.number(), slug: z.string(), name: z.string() })),
  tags: z.array(z.object({ id: z.number(), slug: z.string(), name: z.string() })),
})

export const GameListResponseSchema = z.object({
  items: z.array(GameListItemSchema),
  total: z.number(),
  page: z.number(),
  page_size: z.number(),
})

export type GameRead = z.infer<typeof GameReadSchema>
export type GameListResponse = z.infer<typeof GameListResponseSchema>

File: backend/app/schemas/game.py
import uuid
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict

from app.schemas.taxonomy import GameModeRead, GenreRead, PlatformRead, TagRead


class GameListItem(BaseModel):
    id: uuid.UUID
    title: str
    cover_url: str | None = None
    release_date: datetime | None = None
    igdb_rating: float | None = None
    metacritic_score: int | None = None
    opencritic_score: int | None = None
    steam_score: int | None = None
    steam_id: int | None = None
    hltb_main: float | None = None
    edition_type: Literal["original", "remaster", "remake", "expanded"] = "original"
    parent_game_id: uuid.UUID | None = None
    genres: list[GenreRead] = []
    platforms: list[PlatformRead] = []

    model_config = ConfigDict(from_attributes=True)


class GameRead(GameListItem):
    summary: str | None = None
    storyline: str | None = None
    developer: str | None = None
    steam_description: str | None = None
    steam_reviews_summary: str | None = None
    hltb_extra: float | None = None
    hltb_completionist: float | None = None
    opencritic_excerpts: list[str] | None = None
    keywords: list[str] | None = None
    modes: list[GameModeRead] = []
    tags: list[TagRead] = []


class GameListResponse(BaseModel):
    items: list[GameListItem]
    total: int
    page: int
    page_size: int
